x/gamm/pool-models/concentrated-liquidity/pool.go

```diff
@@ -40,15 +89,85 @@ func (p Pool) SwapOutAmtGivenIn(ctx sdk.Context, tokenIn sdk.Coins, tokenOutDeno
 
 // this only works on a single directional trade, will implement bi directional trade in next milestone
 func (p Pool) CalcOutAmtGivenIn(ctx sdk.Context, tokenIn sdk.Coins, tokenOutDenom string, swapFee sdk.Dec) (tokenOut sdk.Coin, err error) {
-	return sdk.Coin{}, nil
+	tokenAmountInAfterFee := tokenIn[0].Amount.ToDec().Mul(sdk.OneDec().Sub(swapFee))
+
+	// TODO: Replace with spot price
+	priceLower := sdk.NewDec(4500)
+	priceCur := sdk.NewDec(5000)
+	priceUpper := sdk.NewDec(5500)
+
+	sqrtpLow, _ := priceLower.ApproxSqrt()
+	sqrtpLow = sqrtpLow.Mul(sdk.NewDec(10).Power(6))
+	sqrtpCur, _ := priceCur.ApproxSqrt()
+	sqrtpCur = sqrtpCur.Mul(sdk.NewDec(10).Power(6))
+	sqrtpUpp, _ := priceUpper.ApproxSqrt()
+	sqrtpUpp = sqrtpUpp.Mul(sdk.NewDec(10).Power(6))
+
+	// TODO: Roman change out with query to pool to get this info
+	amountETH := int64(1000000)
+	amountUSDC := int64(5000000000)
+
+	liq0 := Liquidity0(amountETH, sqrtpCur, sqrtpUpp)
+	liq1 := Liquidity1(amountUSDC, sqrtpCur, sqrtpLow)
+
+	var liq sdk.Dec
+	if liq0.LT(liq1) {
+		liq = liq0
+	} else {
+		liq = liq1
+	}
+
+	priceDiff := tokenAmountInAfterFee.QuoTruncateMut(liq).Mul(sdk.NewDec(10).Power(6))
```

Comment:
Can we understand why we need to truncate and potentially remove it if it's not needed?